src/Executor/Promise/Adapter/AmpPromiseAdapter.php

```diff
@@ -77,8 +79,13 @@ public function createRejected(Throwable $reason): Promise
         return new Promise($promise, $this);
     }
 
-    public function all(array $promisesOrValues): Promise
+    public function all(iterable $promisesOrValues): Promise
     {
+        assert(
+            is_array($promisesOrValues),
+            'AmpPromiseAdapter::all() requires arg to of array type'
```

Comment:
```suggestion
            'AmpPromiseAdapter::all(): Argument #1 ($promisesOrValues) must be of type array'
```

This is closer to the native PHP error, see https://3v4l.org/dvLAv